src/main/java/uk/gov/companieshouse/company_appointments/service/CompanyAppointmentService.java

```diff
@@ -108,18 +116,14 @@ public OfficerList fetchAppointmentsForCompany(FetchAppointmentsRequest request)
             throw new NotFoundException(String.format("Appointments for company [%s] not found", companyNumber));
         }
 
-        List<OfficerSummary> officerSummaries = allAppointmentData.stream().map(companyAppointmentMapper::map).collect(Collectors.toList());
-        int count = (int) officerSummaries.stream().filter(officer -> officer.getResignedOn() == null).count();
-        int activeCount = 0;
-        int inactiveCount = 0;
-        String appointmentStatus = allAppointmentData.get(0).getCompanyStatus();
-        if (appointmentStatus.equals("removed") || appointmentStatus.equals("dissolved") || appointmentStatus.equals("converted-closed")){
-            inactiveCount = count;
-        } else {
-            activeCount = count;
-        }
+        String companyStatus = allAppointmentData.get(0).getCompanyStatus();
 
-        int resignedCount = (int) officerSummaries.stream().filter(officer -> officer.getResignedOn() != null && officer.getResignedOn().isBefore(ChronoLocalDate.from(LocalDate.now().atStartOfDay()))).count();
+        Counts counts = Optional.ofNullable(metricsApi.getCounts())
+                .map(CountsApi::getAppointments)
+                .map(appointments -> new Counts(appointments, companyStatus))
+                .orElseThrow(() -> new NotFoundException(String.format("Appointments for company [%s] not found", companyNumber)));
```

Comment:
Perhaps we should change this message to something like "Appointments metrics for company number not found"?

Reply:
Done